### PR TITLE
No network error messages in Notifications list

### PR DESCRIPTION
**Pull request by ctarda**
Implements part of #5759 

Expected behaviour:
- If there is no network connection, and there is content available offline, display an alert
- If there is no network connection, and there is no content available offline, present a message in NoResultsView

To test:
1. Enter flight mode
2. Navigate to the Notifications Tab

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift [hunk 1]
```diff
@@ -751,21 +755,41 @@ extension NotificationsViewController {
 
         let start = Date()
 
-        mediator.sync { (error, _) in
+        mediator.sync { [weak self] (error, _) in
```
Comment (elibud) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift]: Extra points for this fix!

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift [hunk 2]
```diff
@@ -1068,12 +1092,22 @@ private extension NotificationsViewController {
...
+        }
+
         // Refresh its properties: The user may have signed into WordPress.com
         noResultsView.titleText     = noResultsTitleText
         noResultsView.messageText   = noResultsMessageText
         noResultsView.buttonTitle   = noResultsButtonText
     }
 
+    private func showNoConnectionView() {
+        noResultsView.titleText     = NSLocalizedString("No notifications yet.", comment: "Displayed in the Notifications Tab as a title, when there are no notifications")
```
Comment (elibud) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift]: Here too, I think the more appropriate title is "Unable to Sync".

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift (hunk 1 above)
Comment (ctarda) [reply WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift]: 😊

**Comment by ctarda** [comment]
And another update!

**Pull request merged**